### DISCUSSION TITLE
How to access model in a table action

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4597] [Help] [answered]
I have trouble wrapping my head around how to access the current row's model and it's data when definig table actions.
I want to insert a link/button with a custom route that generates a pdf.

Here is my table function of my resource (simplified):

```

public static function table(Table $table): Table
{
	return $table
		->columns([
			Tables\Columns\TextColumn::make('number')
				->sortable()
				->searchable()
				->label('Rechnungs-Nr.')
			,
			...

		])		
		->actions([
			Tables\Actions\EditAction::make(),
			// the following does not work
			Tables\Actions\Action::make('print_pdf')
				->label('PDF')
				->url(route('invoices.pdf', ['invoice' => fn (Invoice $invoice) => $invoice->getId()]))
				->color('success'),
		]);
}



```

This results in this error: `Object of class Closure could not be converted to string`.

The underlying question I have: how do I access a model and it's data of the current row? 
This was not more than an educated guess: `fn (Invoice $invoice) => $invoice->getId()`

Thank you

**Comment by OvalMedia** [comment, marked as answer]
This seems to work. At least there is no error anymore. But instead of a linkt with an url a button gets rendered...
`
Tables\Actions\Action::make('print_pdf')
	->label('PDF')
	->url(function(Invoice $record) {
		route('invoices.pdf', ['invoice' => $record->id]);
	})
	->color('success'),

`

**Reply by danharrin** [reply]
Tried passing `->link()` ?

**Reply by OvalMedia** [reply]
Fixed it here: https://github.com/filamentphp/filament/discussions/4600
